### PR TITLE
Added DBAL v3 support

### PR DESCRIPTION
**Pull request by tabacitu**
Depends on #3516 
Fixes #3347

Work in Progress. Looks like in DBAL v3 the way we `setDoctrineTypesMapping` completely stops working...

### CODE REVIEW

**Comment by tabacitu** [comment]
@pxpm any chance you can take over for me here? 👀 I was going to finish it this week but... you know what happened... life happened 😂 You're a lot more familiar with how/why we do `setDoctrineTypesMapping` anyway...

**Comment by pxpm** [comment]
Hello @tabacitu 

After some research I realized `json_array` type had been deprecated: https://github.com/doctrine/dbal/blob/2.12.x/lib/Doctrine/DBAL/Types/JsonArrayType.php

And also `DBALException`. 

I fixed the code to run tests with dbal 3.0, then I downgraded to 2.5 (2.12) and the tests still running with the changes. 

From what I see L8 is passing all tests with dbal 2.5 or 3.0.

< L8 are failing for dbal 3.0 and passing for 2.5. Should not be a problem because L7 is locked at `"doctrine/dbal": "^2.6",`

Let me know.

Pedro

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -27,8 +28,10 @@ jobs:
                         php: 7.2
                     -   laravel: 8.*
                         phpunit: 8.*
+                    -   php: 7.2
+                        dbal: ^3.0
```
Comment (promatik) [inline .github/workflows/testing.yml:33]:
```suggestion
                    -   dbal: ^3.0
                        laravel: 6.*
                    -   dbal: ^3.0
                        laravel: 7.*

```

**Comment by promatik** [comment]
@tabacitu, this PR had commits from you and Pedro, I felt I was missing the party so I joined it.
As @pxpm said, Laravel 6.* and 7.* is locked at DBAL ^2.6, so no need to test it right?

I was actually curious about Github workflow 😅 hope you don't mind.

**Comment by tabacitu** [comment]
Excellent guys! Thanks a lot 🥳

**Pull request merged**